### PR TITLE
feat: support arbitrary mount options for each cache

### LINKED ISSUE #12: Allow setting id of mount
When using docker --mount=type=cache, there is an optional argument of `id`. This can be used to cache the same directory under a different volume to prevent collisions. I'm not sure that it makes a ton of sense to use in this context, but when I was following a guide they provided an id for the mount. However, this action does not accept an id. This was confusing because it led to this action not working (as its injection/extraction steps dont set the id, so it wont match during the real run). I'm not sure how important it is to add, but could be nice for consistency and preventing confusion

### PR DESCRIPTION
**Pull request by aminya**
Fixes #12 
Closes #21

### CODE REVIEW

File: src/opts.ts [hunk 1]
```diff
@@ -45,17 +45,54 @@ Save 'RUN --mount=type=cache' caches on GitHub Actions or other CI platforms
...
+export function getMountArgsString(cacheOptions: CacheOptions): string {
+  if (typeof cacheOptions === "string") {
+    // only the target path is provided
+    return `type=cache,target=${cacheOptions}`;
+  } else {
+    // other options are provided
+    const otherOptions = Object.entries(cacheOptions).map(([key, value]) => `${key}=${value}`).join(",");
+    return `type=cache,${otherOptions}`;
+  }
+}
```
Comment (AkihiroSuda) [inline src/opts.ts:98]: Maybe we should consider having unit tests for these functions

Comment (aminya) [reply src/opts.ts:98]: I added a test for it in the CI.

Comment (aminya) [reply src/opts.ts:99]: Added unit tests for the opts functions in [`d2e9ac7` (#28)](https://github.com/reproducible-containers/buildkit-cache-dance/pull/28/commits/d2e9ac74638c470b0ca1ba99c9d21628d7aeb058)

**Comment by aminya** [comment]
@AkihiroSuda Any more comment on this?

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by aminya** [comment]
You're welcome! It's cool to see another robotics engineer using these tools.

 I find the path parameter usually enough for my use cases, but feel free to document your findings in the readme.

**Pull request merged**